src/apis/watch.ts

```diff
@@ -188,8 +198,13 @@ function createWatcher(
       before: runCleanup,
     });
 
+    patchWatcherTeardown(watcher, runCleanup);
+
     // enable the watcher update
     watcher.lazy = false;
+    if (vm !== fallbackVM) {
+      vm._watchers.push(watcher);
```

Comment:
Why are you pushing the watch again to the vm?
```ts
// vm._watchers[0] === vm._watchers[1]: true
```
[createVueWatcher](https://github.com/vuejs/composition-api/blob/fix-293-watch_cleanup-on-unmount/src/apis/watch.ts#L117-L140) already pushes the watcher to `vm._watchers` and returns the last

Reply:
good catch, will remove that.

Reply:
done